tool/tsh/tsh_test.go

```diff
@@ -498,18 +497,10 @@ func TestMakeClient(t *testing.T) {
 // approveAllAccessRequests starts a loop which gets all pending AccessRequests
 // from access and approves them. It accepts a stop channel, which will stop the
 // loop and cancel all active requests when it is closed.
-func approveAllAccessRequests(access services.DynamicAccess, stop <-chan struct{}) (err error) {
+func approveAllAccessRequests(ctx context.Context, access services.DynamicAccess) (err error) {
 	ticker := time.NewTicker(500 * time.Millisecond)
 	defer ticker.Stop()
 
-	// Make a context that will be cancelled when the stop chan is closed, to
-	// pass to the requests.
-	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
Nice work pulling this weird bit of code out and just replacing it with a ctx passed in.

Reply:
heh my own weird code from a few days ago... was going to be a type with a couple methods and ended up being a single function... coming back to something with fresh eyes can be great